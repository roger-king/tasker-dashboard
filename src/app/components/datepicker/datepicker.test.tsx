import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from './index';

describe('<DatePicker />', () => {
    it('Matches the snapshot with datepicker closed', () => {
        const onSelect = jest.fn();
        const setShowCalendar = jest.fn();
        const datepicker = renderer
            .create(
                <DatePicker date={null} onSelect={onSelect} setShowCalendar={setShowCalendar} showCalendar={false} />,
            )
            .toJSON();
        expect(datepicker).toMatchSnapshot();
    });

    it('Matches the snapshot with datepicker open', () => {
        const onSelect = jest.fn();
        const setShowCalendar = jest.fn();
        const datepicker = renderer
            .create(<DatePicker date={null} onSelect={onSelect} setShowCalendar={setShowCalendar} showCalendar />)
            .toJSON();
        expect(datepicker).toMatchSnapshot();
    });
});
