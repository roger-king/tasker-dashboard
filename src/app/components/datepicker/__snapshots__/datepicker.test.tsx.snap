// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> Matches the snapshot with datepicker closed 1`] = `
<button
  aria-label="Open Drop"
  className="StyledButton-sc-323bzc-0 jfbKsv"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bLOKyd"
  >
    <span
      className="StyledText-sc-1sadyjn-0 fWSbXS"
    >
      12/1/2019
    </span>
    <div
      className="StyledBox-sc-13pk1d4-0 bspmCs"
    >
      <svg
        aria-label="Calendar"
        className="StyledIcon-ofa7kd-0 btwGsx"
        viewBox="0 0 24 24"
      >
        <path
          d="M2,5 L22,5 L22,22 L2,22 L2,5 Z M18,5 L18,1 M6,5 L6,1 M2,10 L22,10"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </div>
  </div>
</button>
`;

exports[`<DatePicker /> Matches the snapshot with datepicker open 1`] = `
<button
  aria-label="Open Drop"
  className="StyledButton-sc-323bzc-0 jfbKsv"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bLOKyd"
  >
    <span
      className="StyledText-sc-1sadyjn-0 fWSbXS"
    >
      12/1/2019
    </span>
    <div
      className="StyledBox-sc-13pk1d4-0 bspmCs"
    >
      <svg
        aria-label="Calendar"
        className="StyledIcon-ofa7kd-0 btwGsx"
        viewBox="0 0 24 24"
      >
        <path
          d="M2,5 L22,5 L22,22 L2,22 L2,5 Z M18,5 L18,1 M6,5 L6,1 M2,10 L22,10"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </div>
  </div>
</button>
`;
